internal/types/subtypes/source_registry.go

```diff
@@ -35,7 +35,7 @@ type sourceRegistry struct {
 	m map[protoreflect.FullName]protoreflect.FieldDescriptor
 }
 
-func (s sourceRegistry) register(d protoreflect.MessageDescriptor) error {
+func (s *sourceRegistry) register(d protoreflect.MessageDescriptor) error {
```

Comment:
:thinking: is this needed?

Reply:
Yep, the mutex is otherwise copied and provides no mutual exclusion :P